File: programming-concepts/good-programming-practice/c-sharp/identifiers.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class GoodProgrammingPractice
    {
        const double GravityOnEarth = 9.81;
        const double GravityOnMars = 3.711;


        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            // Execute both subroutines and output the results
            double weight = 35.7;
            Console.WriteLine(Conv(weight));
            Console.WriteLine(CalculateWeightOnMars(weight));
        }


        // First example
        public static double Conv(double a)
        {
            double b = a / 9.81 * 3.711;
            return b;
        }


        // Second example
        public static double CalculateWeightOnMars(double weightOnEarth)
        {
            double weightOnMars = weightOnEarth / GravityOnEarth * GravityOnMars;
            return weightOnMars;
        }


    }
}

File: programming-concepts/good-programming-practice/python/identifiers.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4


# First example
def conv(a):
    b = a / 9.81 * 3.711
    return b
# These identifiers do not help the programmer understand what is happening in the program


# Second example
GRAVITY_ON_EARTH = 9.81
GRAVITY_ON_MARS = 3.711


# This function takes a float as its parameter, and returns a float
def calculate_weight_on_mars(weight_on_earth: float) -> float:
    weight_on_mars = (weight_on_earth / GRAVITY_ON_EARTH) * GRAVITY_ON_MARS
    return weight_on_mars
# These identifiers clearly show the meaning and purpose of the data they hold.


def main():
    """
    Execute both subroutines and output the results
    """

    weight = 35.7
    print(conv(weight))
    print(calculate_weight_on_mars(weight))
    # It is possible to specify parameters for clarity: calculate_weight_on_mars(weight_on_earth=weight)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
